codexctl/analysis.py

```
@@ -0,0 +1,33 @@
+    image = UpdateImage(file)
+    volume = ext4.Volume(image, offset=0)
+    try:
+        inode = volume.inode_at("/usr/share/update_engine/update-payload-key.pub.pem")
+        if inode is None:
+            raise FileNotFoundError()
+
+        inode.verify()
+        image.verify(inode.open().read())
+
+    except UpdateImageSignatureException:
+        warnings.warn("Signature doesn't match contents", RuntimeWarning)
```

Ruff (0.8.0)

No explicit `stacklevel` keyword argument found

(B028)